instrumentation/opentelemetry-instrumentation-boto3sqs/src/opentelemetry/instrumentation/boto3sqs/__init__.py

```diff
@@ -0,0 +1,428 @@
+    MessagingDestinationKindValues,
+    MessagingOperationValues,
+    SpanAttributes,
+)
+from opentelemetry.trace import Link, Span, SpanKind, Tracer, TracerProvider
+
+from .package import _instruments
+from .version import __version__
+
+logger = logging.getLogger(__name__)
+# We use this prefix so we can request all instrumentation MessageAttributeNames with a wildcard, without harming
+# existing filters
+OPENTELEMETRY_ATTRIBUTE_IDENTIFIER: str = "otel."
+
+
+class Boto3SQSGetter(Getter):
+    def get(self, carrier: CarrierT, key: str) -> Optional[List[str]]:
+        value = carrier.get(f"{OPENTELEMETRY_ATTRIBUTE_IDENTIFIER}{key}", {})
+        if not value:
+            return None
+        return [value.get("StringValue", None)]
+
+    def keys(self, carrier: CarrierT) -> List[str]:
+        return [
+            key[len(OPENTELEMETRY_ATTRIBUTE_IDENTIFIER) :]
+            if key.startswith(OPENTELEMETRY_ATTRIBUTE_IDENTIFIER)
+            else key
+            for key in carrier.keys()
+        ]
+
+
+class Boto3SQSSetter(Setter):
+    def set(self, carrier: CarrierT, key: str, value: str) -> None:
+        # This is a limitation defined by AWS for SQS MessageAttributes size
+        if len(carrier.items()) < 10:
+            carrier[f"{OPENTELEMETRY_ATTRIBUTE_IDENTIFIER}{key}"] = {
+                "StringValue": value,
+                "DataType": "String",
+            }
+        else:
+            logger.warning(
+                "Boto3 SQS instrumentation: cannot set context propagation on SQS/SNS message due to maximum amount of "
+                "MessageAttributes"
+            )
+
+
+boto3sqs_getter = Boto3SQSGetter()
+boto3sqs_setter = Boto3SQSSetter()
+
+
+# pylint: disable=attribute-defined-outside-init
+class Boto3SQSInstrumentor(BaseInstrumentor):
+    received_messages_spans: Dict[str, Span] = {}
+    current_span_related_to_token: Span = None
+    current_context_token = None
+
+    class ContextableList(list):
+        """
+        Since the classic way to process SQS messages is using a `for` loop, without a well defined scope like a
+        callback - we are doing something similar to the instrumentaiton of Kafka-python and instrumenting the
+        `__iter__` functions and the `__getitem__` functions to set the span context of the addressed message. Since
+        the return value from an `SQS.ReceiveMessage` returns a builtin list, we cannot wrap it and change all of the
+        calls for `list.__iter__` and `list.__getitem__` - therefore we use ContextableList. It is bound to the
+        received_messages_spans dict
+        """
+
+        def __getitem__(
+            self, *args: List[Any], **kwargs: Dict[str, Any]
+        ) -> Any:
+            retval = super(
+                Boto3SQSInstrumentor.ContextableList, self
+            ).__getitem__(*args, **kwargs)
+            if not isinstance(retval, dict):
+                return retval
+            receipt_handle = retval.get("ReceiptHandle", None)
```

Comment:
Nit: `None` is the default return value of `dict.get`:
```suggestion
            receipt_handle = retval.get("ReceiptHandle")
```

Reply:
Done